### ISSUE TITLE
Why validation with lambda if/unless are not documented?

### ISSUE THREAD

**Issue by Ana06** [rails/rails#33212]
In the [validation documentation](http://guides.rubyonrails.org/active_record_validations.html#conditional-validation), it is said that Symbol and Proc can be used in the `if`/`unless` option of a validation to decide when the validation is executed. But lambas can also be used, for example:

```ruby
validates :password, confirmation: true, unless: -> { password.blank? }
```

Is there any reason why this is not documented? :thinking:

**Comment by Ana06** [comment]
@rafaelfranca that's true... Thanks for the answer! :wink:

**Comment by rafaelfranca** [comment]
If you had doubts I'm sure other people can have it. What do you think about changing one of the examples in that page to use a lambda?

**Comment by sikachu** [comment]
I agree! Please feel free to make a documentation patch to show that one can use lambda. We're looking forward to see your patch.

**Comment by Ana06** [comment]
ok, I'll send a PR then :wink:

**Comment by sergey-alekseev** [comment]
>Lambdas are also Procs.

What doesn't really guarantee lambdas are supported. See 1cec84ad2ddd843484ed40b1eb7492063ce71baf. So it's good to be explicit 👍
